packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts

```diff
@@ -130,8 +130,12 @@ async function processDocumentReference({
       docUniqueId: documentResponse.DocumentUniqueId.toString(),
       metriportId: metriportId,
       fileLocation: bucket,
-      homeCommunityId: stripUrnPrefix(documentResponse.HomeCommunityId),
-      repositoryUniqueId: documentResponse.RepositoryUniqueId,
+      homeCommunityId: documentResponse.HomeCommunityId
+        ? stripUrnPrefix(documentResponse.HomeCommunityId)
+        : outboundRequest.gateway.homeCommunityId,
+      repositoryUniqueId: documentResponse.RepositoryUniqueId
+        ? stripUrnPrefix(documentResponse.RepositoryUniqueId)
+        : outboundRequest.gateway.homeCommunityId,
```

Comment:
using request home community id in the case where we didnt get one

Reply:
Just to make sure it's not a copy/paste typo, for `repositoryUniqueId` we also fallback to the request's `homeCommunityId`, right?